src/PolicyEditor.js

```diff
@@ -496,6 +535,116 @@ export class PolicyEditor extends Component {
                             </Row>
                         </Accordion.Body>
                     </Accordion.Item>
+                    <Accordion.Item eventKey="snapshot-actions">
+                        <Accordion.Header><FontAwesomeIcon icon={faFileAlt} />&nbsp;Snapshot Actions</Accordion.Header>
+                        <Accordion.Body>
+                            <SectionHeaderRow />
+                            <Row>
+                                <LabelColumn name="Before Snapshot" help="Script to run before snapshot." />
+                                <WideValueColumn>{OptionalFieldNoLabel(this, "", "policy.actions.beforeSnapshotRoot.path", {})}</WideValueColumn>
+                                {EffectiveValue(this, "actions.beforeSnapshotRoot.path")}
+                            </Row>
+                            <Row>
+                                <LabelColumn name="Timeout" help="Timeout in seconds before Kopia kills the process." />
+                                <WideValueColumn>{OptionalNumberField(this, "", "policy.actions.beforeSnapshotRoot.timeout", {})}</WideValueColumn>
+                                {EffectiveValue(this, "actions.beforeSnapshotRoot.timeout")}
+                            </Row>
+                            <Row>
+                                <LabelColumn name="Command Mode" help="essential (must succeed, default behavior), optional (failures are tolerated) or async (kopia will start the action but not wait for it to finish)." />
+                                <WideValueColumn>
+                                    <Form.Control as="select" size="sm"
+                                        name="policy.actions.beforeSnapshotRoot.mode"
+                                        onChange={this.handleChange}
+                                        value={stateProperty(this, "policy.actions.beforeSnapshotRoot.mode")}>
+                                        <option value="essential">essential</option>
```

Comment:
can we move per-item explanation from label to options here?

```
<option value="essential">must succeed</option>
<option value="optional">ignore failures</option>
<option value="async">run asynchronously, ignore failures</option>
```

Reply:
Done (multiple times for now, see above)